cloudinit/stages.py

```diff
@@ -710,26 +751,35 @@ def _apply_netcfg_names(self, netcfg):
             LOG.warning("Failed to rename devices: %s", e)
 
     def apply_network_config(self, bring_up):
-        # get a network config
+        """Apply the network config.
+
+        Find the config, determine whether to apply it, apply it via
+        the distro, and optionally bring it up
+        """
         netcfg, src = self._find_networking_config()
         if netcfg is None:
             LOG.info("network config is disabled by %s", src)
             return
 
-        # request an update if needed/available
-        if self.datasource is not NULL_DATA_SOURCE:
-            if not self.is_new_instance():
-                if not self.datasource.update_metadata([EventType.BOOT]):
-                    LOG.debug(
-                        "No network config applied. Neither a new instance"
-                        " nor datasource network update on '%s' event",
-                        EventType.BOOT)
-                    # nothing new, but ensure proper names
-                    self._apply_netcfg_names(netcfg)
-                    return
-                else:
-                    # refresh netcfg after update
-                    netcfg, src = self._find_networking_config()
+        if not (
+            self.datasource is NULL_DATA_SOURCE or
+            self.boot_type == EventType.BOOT_NEW_INSTANCE or
+            (
+                self.update_event_enabled(
+                    self.boot_type, scope=EventScope.NETWORK
+                ) and self.datasource.update_metadata([self.boot_type])
+            )
+        ):
```

Comment:
I'm having a hard time following the exact logic here; could we decompose it into one or more variables, that'll make it a little easier to understand what we're checking?